src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -408,6 +408,9 @@ public HdAccount GetUnusedAccount(Wallet wallet, string password)
 
                 // No unused account was found, create a new one.
                 account = wallet.AddNewAccount(password, this.coinType, this.dateTimeProvider.GetTimeOffset());
+                IEnumerable<HdAddress> newReceivingAddresses = account.CreateAddresses(this.network, this.walletSettings.UnusedAddressesBuffer);
```

Comment:
Won't it be better to move this "creation" logic into `AddNewAccount`?

Reply:
In theory yes. However, sometimes (like in cold staking), you want to just create an account and be in control of what you generate.

Reply:
OK makes sense, I'm just worried that if someone calls `AddNewAccount` from anywhere and expect addresses to be in there, there won't be?